warehouse/query-core/src/main/java/datawave/query/transformer/UniqueTransform.java

```diff
@@ -95,279 +69,51 @@ public void updateConfig(UniqueFields uniqueFields, QueryModel model) {
     }
     
     /**
-     * Get a predicate that will apply this transform.
-     *
-     * @return A unique transform predicate
-     */
-    public Predicate<Entry<Key,Document>> getUniquePredicate() {
-        return input -> UniqueTransform.this.apply(input) != null;
-    }
-    
-    /**
-     * Apply uniqueness to a document.
-     *
+     * Apply uniqueness to a document and if it is unique, stores it in the internal list. Null is returned from this method. #flush() should be called in order
+     * to retrieve the next .
+     * 
      * @param keyDocumentEntry
-     * @return The document if unique per the configured fields, null otherwise.
+     * @return null
      */
     @Nullable
     @Override
     public Entry<Key,Document> apply(@Nullable Entry<Key,Document> keyDocumentEntry) {
         if (keyDocumentEntry != null) {
-            if (FinalDocumentTrackingIterator.isFinalDocumentKey(keyDocumentEntry.getKey())) {
-                return keyDocumentEntry;
-            }
-            
             try {
-                if (isDuplicate(keyDocumentEntry.getValue())) {
-                    keyDocumentEntry = null;
+                if (!FinalDocumentTrackingIterator.isFinalDocumentKey(keyDocumentEntry.getKey())
+                                && uniqueUtil.isDuplicate(keyDocumentEntry.getValue(), uniqueFields, modelMapping)) {
+                    return null;
                 }
             } catch (IOException ioe) {
                 log.error("Failed to convert document to bytes.  Returning document as unique.", ioe);
             }
+            documents.add(keyDocumentEntry);
```

Comment:
Not using a list (and the flush() method) will cause one result to be dropped each time an intermediate result is hit.  This is because an intermediate result is "replaced" with an empty document, and without a backing list, that result that was "replaced" is lost.  The `RunnigQuery` call itself does not have a way to persist that result across calls to the next()  -- with our current implementation, a list needs to be kept somewhere, or else we will lose results each time we hit an intermmediate result. 

This is reproducible by commenting out line 112 and lines 31 - 136 of  the `testLongRunningUniqueQuery`.  Commenting out those lines turns off the intermediate result functionality, and the relevant assertions.  You will see that you get 3 results.  

Changing the code in `UniqueTransform` to what is suggested, specifically the following, and uncommenting out the above lines and re-running the test, you will see that you get an assertion error, and only 2 results. 
```
public Entry<Key,Document> apply(@Nullable Entry<Key,Document> keyDocumentEntry) {
        if (keyDocumentEntry != null) {
            try {
                if (!FinalDocumentTrackingIterator.isFinalDocumentKey(keyDocumentEntry.getKey())
                                && uniqueUtil.isDuplicate(keyDocumentEntry.getValue(), uniqueFields, modelMapping)) {
                    return null;
                }
            } catch (IOException ioe) {
                log.error("Failed to convert document to bytes.  Returning document as unique.", ioe);
            }
//            documents.add(keyDocumentEntry);
        }
        
        long elapsedExecutionTimeForCurrentPage = System.currentTimeMillis() - this.queryExecutionForPageStartTime;
        if (elapsedExecutionTimeForCurrentPage > this.queryExecutionForPageTimeout) {
            Document document = new Document();
            document.setIntermediateResult(true);
            return Maps.immutableEntry(new Key(), document);
        }
        
        return keyDocumentEntry;
    }
    
    /**
     * Returns the next unique document in the pipeline, if there is one. Also checks to see if the timeout for the page has exceeded and interjects an
     * intermediate result instead of hte next unique document if so.
     *
     * @return The next unique document per the configured fields, or an intermediate result (blank document) if the page execution time has elapsed, or null if
     *         there isn't one and the timeout hasn't elapsed.
     */
//    @Override
//    public Entry<Key,Document> flush() {
//        Entry<Key,Document> entry = null;
//        if (!documents.isEmpty()) {
//            entry = documents.pop();
//        }
//
//        return entry;
//    }
```

While I understand this is less than ideal because the apply() and flush() mechanism waits until all results are ready, before sending them to the client as opposed to streaming them,  I don't think changing this to a true streaming paradigm can currently be done without major modifications, chiefly to the `RunningQuery` to make it aware of result state across calls to `next()`   Therefore, I did not make this change, nor the one above.